File: trade-champ/frontend/types/trade.ts
export type Trade = {
  id: number;
  ticket: number;
  symbol: string;
  strategy_tag: string | null;
  profit: number;
  open_time: string;
  entry_price: number;
  exit_price: number | null;
  stop_loss: number | null;
  take_profit: number | null;
  trade_duration_minutes: number | null;
  notes?: string;
  screenshot_url?: string;
};

File: trade-champ/backend/app/schemas/trade.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class Trade(BaseModel):
    id: int
    user_id: str
    ticket: int
    symbol: str
    volume: float
    entry_price: float
    exit_price: Optional[float] = None
    stop_loss: Optional[float] = None
    take_profit: Optional[float] = None
    open_time: datetime
    close_time: Optional[datetime] = None
    profit: float
    commission: float = 0
    swap: float = 0
    trade_duration_minutes: Optional[int] = None
    strategy_tag: Optional[str] = None
